### PR TITLE
Enable Javascript bindings for photo module.

### LINKED ISSUE #13299: OpenCV.js : build with "photo" module
I am working on demo-ing some new camera functionalities like exposureTime and focusDistance which I have recently upstreamed to Chrome browser. I would like to create a HDR image (capture, process, display) on browser, so I need the "photo" module bindings in Javascript.

Link to the WIP demo page:
https://github.com/riju/WebCamera/tree/master/samples/hdr

### PR DESCRIPTION
**Pull request by riju**
resolves #13299 

1. Enable the build flag in build_js.py.
2. Append js into WRAP list of photo’s CMakefiles.txt
3. Add photo module's API into JS API whitelist (embindgen.py)

TODO:
- [x] Add tests
- [ ] Fix opencv/doc/js_tutorials/

```
**WIP**
force_builders=Docs,Custom
docker_image:Docs=docs-js
docker_image:Custom=javascript
buildworker:Custom=linux-1,linux-2,linux-4
```

### CODE REVIEW

**Comment by alalek** [comment]
Thank you! Looks good to me.

Could you please add some simple tests to verify that bindings work?

Documentation snapshot (with JS) with this patch is here: http://pullrequest.opencv.org/buildbot/export/pr/13300/docs/

**Comment by riju** [comment]
@huningxin : Looks like I am missing something to properly expose these functions for HDR, as evident from the small test file. Any ideas ?

`   at Function.Module._load (module.js:497:3): cv.createMergeDebevec is not a function
TypeError: cv.createMergeDebevec is not a function
    at Object.<anonymous> (/home/riju/opencv/build_js/bin/test_photo.js:61:24)
`

File: modules/js/src/embindgen.py [hunk 1]
```diff
@@ -142,6 +142,21 @@
...
+              'createTonemapDrago',
+              'createTonemapMantiuk', 'createTonemapReinhard'],
+        'AlignMTB' : ['calculateShift', 'shiftMat', 'computeBitmaps', 'getMaxBits', 'setMaxBits', 'getExcludeRange', 'setExcludeRange', 'getCut', 'setCut'],
+        'CalibrateDebevec' : ['getLambda', 'setLambda', 'getSamples', 'setSamples', 'getRandom', 'setRandom'],
+        'CalibrateRobertson' : ['getMaxIter', 'setMaxIter', 'getThreshold', 'setThreshold', 'getRadiance'],
+        'MergeDebevec' : ['process'],
+        'MergeMertens' : ['process', 'getContrastWeight', 'setContrastWeight', 'getSaturationWeight', 'setSaturationWeight', 'getExposureWeight', 'setExposureWeight'],
+        'MergeRobertson' : ['process'],
+        'TonemapDrago' : ['getSaturation', 'setSaturation', 'getBias', 'setBias'],
+        'getSigmaColor', 'setSigmaColor', 'getSigmaSpace', 'setSigmaSpace'],
```
Comment (huningxin) [inline modules/js/src/embindgen.py]: this line fails the build, could you please double check?

Comment (riju) [reply modules/js/src/embindgen.py]: thanks, I fixed that now. I tried with commenting out everything apart from `createCalibrateDebevec`., just to check exposing of 1 single function. Still not exposed.

File: modules/js/test/test_photo.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    var cv = require('./opencv.js'); // eslint-disable-line no-var
+}
+
+
+QUnit.module('Photo', {});
+
+QUnit.test('test_photo', function(assert) {
+    // createCalibrateDebevec
+    {
+        let calibration = cv.createCalibrateDebevec();
```
Comment (huningxin) [inline modules/js/test/test_photo.js]: this should call the constructor like `let calibration = new cv.CalibrateDebevec()`

File: modules/js/src/embindgen.py [hunk 3]
```diff
@@ -142,6 +142,27 @@
...
 
+photo = {'': [#'createAlignMTB',
+              'createCalibrateDebevec',
+              #'createCalibrateRobertson', \
+              #'createMergeDebevec', 'createMergeMertens', 'createMergeRobertson', \
+              #'createTonemapDrago', 'createTonemapMantiuk', 'createTonemapReinhard'
+              ],
+        #'AlignMTB' : ['calculateShift', 'shiftMat', 'computeBitmaps', 'getMaxBits', 'setMaxBits', \
+        #              'getExcludeRange', 'setExcludeRange', 'getCut', 'setCut'],
+        'CalibrateDebevec' : ['getLambda', 'setLambda', 'getSamples', 'setSamples', 'getRandom', 'setRandom'],
```
[the remark this answers is not in the document]
Comment (riju) [reply modules/js/src/embindgen.py]: thanks!

**Pull request merged**